aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -19,11 +19,130 @@
 #include <ATen/cuda/detail/KernelUtils.h>
 #include <ATen/native/cuda/Loops.cuh>
 #include <ATen/native/cuda/DistributionTemplates.h>
+#include <ATen/native/Resize.h>
 #include <c10/cuda/CUDAMathCompat.h>
 
 namespace at {
 namespace native {
 
+// -----------------------------------
+// glu forward
+// -----------------------------------
+void glu_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND2(kHalf, kBFloat16, iter.dtype(), "glu_cuda", [&]() {
+    using acc_t = at::acc_type<scalar_t, true>;
+    gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a_, scalar_t b_) -> scalar_t {
+      const acc_t a = a_;
+      const acc_t b = b_;
+      const acc_t one = acc_t(1);
+      const acc_t sigmoid = one / (one + std::exp(-b));
+      return a * sigmoid;
+    });
+  });
+}
+
+// -----------------------------------
+// glu backward
+// -----------------------------------
+
+// Byte offsets don't require multiplication by sizeof(T), so are slightly cheaper.
+// For fixed offsets, this removes all penalty from 64-bit indexing.
+template <typename T>
+__device__ T* byte_offset(T* ptr, int64_t offset) {
+  using byte_ptr_t = typename std::conditional<
+    std::is_const<T>::value, const char*, char*>::type;
+  return reinterpret_cast<T*>(
+    reinterpret_cast<byte_ptr_t>(ptr) + offset
+  );
+}
+
+template <typename scalar_t, typename OffsetCalc>
+__global__ void glu_backward_kernel(
+    int numel, scalar_t* gI, const scalar_t* I, const scalar_t* gO,
+    OffsetCalc offset_calculator,
+    int64_t gI_byte_offset, int64_t I_byte_offset) {
+  using acc_t = at::acc_type<scalar_t, true>;
+  // We explicitly iterate over the first half of the input tensor, and
+  // gI_byte_offset and I_byte_offset are the offsets to access the
+  // corresponding index in the second half of the tensor.
+  CUDA_KERNEL_LOOP(i, numel) {
```

Comment:
nit: you don't need CUDA_KERNEL_LOOP here as you have num_threads >= num_elements